packages/completer-extension/src/index.ts

```diff
@@ -316,6 +317,14 @@ const inlineCompleter: JupyterFrontEndPlugin<void> = {
                       provider.name
                     ),
                     type: 'boolean'
+                  },
+                  fillInMiddle: {
+                    title: trans.__('Fill in middle'),
+                    description: trans.__(
+                      'Whether to show completions in the middle of the code line from %1 provider.',
+                      provider.name
+                    ),
```

Comment:
I like this. It is a better choice over placing this as an option for the entire inline completer (like [these](https://github.com/jupyterlab/jupyterlab/blob/b057cc8d2b12bfaa84470c5ce536c8dd2b8277dd/packages/completer-extension/schema/inline-completer.json#L43-L102)) because:
- some providers will support fill-in-the-middle and some won't
- providers can set a default for themselves (declare support for fill-in-the-middle)

It looks like this is a generic option for both `auto` and `manual` trigger. This can be problematic for two reasons:
- manual trigger is already allowed in the middle of the line; this PR as-is would effectively disable it by default, breaking some existing use cases
- there may be providers which have some support for fill-in-the-middle, but which is not great; in these cases users may want to be able to trigger it manually in the middle of the line but may not want it to show up automatically.

What do you think about the following:

```suggestion
                  autoFillInMiddle: {
                    title: trans.__('Fill in middle on typing'),
                    description: trans.__(
                      'Whether to show completions in the middle of the code line from %1 provider on typing.',
                      provider.name
                    ),
```